File: leetcode/code/412.fizz_buzz.py
class Solution(object):
    def fizzBuzz(self, n):
        result = []

        for i in range(1, n + 1):
            if i % 15 == 0:
                result.append("FizzBuzz")
            elif i % 5 == 0:
                result.append("Buzz")
            elif i % 3 == 0:
                result.append("Fizz")
            else:
                result.append(str(i))

        return result


if __name__ == "__main__":
    assert Solution().fizzBuzz(3) == ["1", "2", "Fizz"]
    assert Solution().fizzBuzz(5) == ["1", "2", "Fizz", "4", "Buzz"]

File: leetcode/code/412.fizz_buzz.go
package main

import (
    "fmt"
    "strconv"
)

func fizzBuzz(n int) []string {
    var result = make([]string, 0)

    for i := 1; i <= n; i++ {
        if i % 15 == 0 {
            result = append(result, "FizzBuzz")
        } else if i % 5 == 0 {
            result = append(result, "Buzz")
        } else if i % 3 == 0 {
            result = append(result, "Fizz")
        } else {
            result = append(result, strconv.Itoa(i))
        }
    }

    return result
}


func main() {
    expected := []string{"1", "2", "Fizz", "4", "Buzz"}
    result := fizzBuzz(5)

    for i := 0; i < 5; i ++ {
        if expected[i] != result[i] {
            fmt.Println("buggy")
        }
    }
}
